app/server/models/transfer_account.py

```diff
@@ -188,39 +188,30 @@ def get_approval(self, receiving_address):
 
     def approve_and_disburse(self, initial_disbursement=None):
         from server.utils.access_control import AccessControl
-
-        active_org = getattr(g, 'active_organisation', self.primary_user.default_organisation)
         admin = getattr(g, 'user', None)
-        auto_resolve = initial_disbursement == active_org.default_disbursement
 
         if not self.is_approved and admin and AccessControl.has_sufficient_tier(admin.roles, 'ADMIN', 'admin'):
             self.is_approved = True
 
         if self.is_beneficiary:
-            # TODO: make this more robust
-            # approve_and_disburse might be called for a second time to disburse
-            # so first check that no credit transfer have already been received
-            if len(self.credit_receives) < 1:
```

Comment:
i can't see where this logic is re-implemented?

Reply:
I checked the code and `approve_and_disburse` can only really be called once unless someone _manually_ set `is_approved` to False.

Reply:
Additionally, the approval function is broken on master branch. This fixes and this simplifies the process.

Reply:
hmmmm i might review this overall.